File: verification/screenshot.ts
import { chromium } from 'playwright';

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  await page.setViewportSize({ width: 1280, height: 1200 });
  await page.goto('http://localhost:3000');

  // Accept policy
  await page.click('button:has-text("I Understand and Accept")');
  await page.waitForTimeout(1000);

  await page.screenshot({ path: 'verification/homepage.png', fullPage: true });

  await page.goto('http://localhost:3000/#/news');
  await page.waitForTimeout(2000);
  await page.screenshot({ path: 'verification/newsview.png', fullPage: true });

  await page.goto('http://localhost:3000/#/notadmin');
  await page.waitForTimeout(5000);
  await page.screenshot({ path: 'verification/adminpanel.png', fullPage: true });

  await browser.close();
})();

File: verification/screenshot.py
from playwright.sync_api import sync_playwright, expect
import time

def run_screenshots():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        page.set_viewport_size({"width": 1280, "height": 1200})

        # Go to home page
        page.goto("http://localhost:3000")

        # Accept policy if modal appears
        try:
            accept_button = page.locator('button:has-text("I Understand and Accept")')
            if accept_button.is_visible(timeout=5000):
                accept_button.click()
        except:
            pass

        time.sleep(2)
        page.screenshot(path="verification/screenshots/homepage.png")

        # Go to Admin Panel
        page.goto("http://localhost:3000/#/notadmin")
        time.sleep(5) # Wait for charts
        page.screenshot(path="verification/screenshots/admin_panel.png", full_page=True)

        # Go to Jobs
        page.goto("http://localhost:3000/#/jobs")
        time.sleep(2)
        page.screenshot(path="verification/screenshots/jobs.png")

        browser.close()

if __name__ == "__main__":
    run_screenshots()
